gateway/src/acme.rs

```diff
@@ -150,15 +142,27 @@ impl AcmeClient {
             AcmeClientError::CertificateSigning
         })?;
 
-        let certificate_chain = order
-            .finalize(&signing_request, &state.finalize)
-            .await
-            .map_err(|error| {
-                error!(%error, "failed to finalize certificate request");
-                AcmeClientError::OrderFinalizing
+        order.finalize(&signing_request).await.map_err(|error| {
+            error!(%error, "failed to finalize certificate request");
+            AcmeClientError::OrderFinalizing
+        })?;
+
+        // Poll for certificate, do this for few rounds.
+        let mut res: Option<String> = None;
+        let mut retries = MAX_RETRIES_CERTIFICATE_FETCHING;
+        while res.is_none() && retries > 0 {
+            res = order.certificate().await.map_err(|error| {
+                error!(%error, "failed to fetch the certificate chain");
+                AcmeClientError::CertificateCreation
             })?;
+            retries -= 1;
+            sleep(Duration::new(1, 0)).await;
```

Comment:
```suggestion
            sleep(Duration::from_secs(1)).await;
```
Just like how this duration time is clearer without having to look up the docs

Reply:
I am not sure what you mean

Reply:
Pieter's suggestion is just more readable since you don't need to look up what the args to `Duration::new` are, and it's how we do it elsewhere in the codebase.

Reply:
Ah, right. Didn't notice he change the method to reduce the args count. Makes sense, will do! Thanks guys! 🤘